src/inmanta/ast/attribute.py

```diff
@@ -119,6 +120,9 @@ def get_new_result_variable(self, instance: "Instance", queue: QueueScheduler) -
         else:
             out = ResultVariable()
 
+        node_ref: Optional[dataflow.InstanceNodeReference] = instance.instance_node
+        if node_ref is not None:
+            out.set_dataflow_node(node_ref.top_node().register_attribute(self.name))
```

Comment:
The setting of the node or the condition?
- setting the node: to bind a node to the newly created `ResultVariable`. For simple variables this happens in the `ExecutionContext` where they live, at creation. For attributes this needs to be assigned at creation as well.
- the condition: we only want to set this if the dataflow representation is enabled, thus if the instance itself has a node connected to it.

Reply:
So this will vanish when resolution is unfied?

Reply:
I don't think so, unifying resolution just means we can look for a `ResultVariable` and then fetch its associated node instead of looking for a node directly. So the associated node still needs to be set.